src/daemon.ts

```diff
@@ -23,87 +24,118 @@ export class Daemon implements AsyncIterable<DenonEvent> {
     this.#config = denon.config; // just as a shortcut
   }
 
-  /**
-   * Restart current process.
-   */
-  private async reload() {
+  /** Restart current process. */
+  private async reload(): Promise<void> {
     if (this.#config.logger && this.#config.logger.fullscreen) {
-      log.debug("clearing screen");
+      logger.debug("clearing screen");
       console.clear();
     }
 
     if (this.#config.watcher.match) {
-      log.info(`watching path(s): ${this.#config.watcher.match.join(" ")}`);
+      logger.info(`watching path(s): ${this.#config.watcher.match.join(" ")}`);
     }
     if (this.#config.watcher.exts) {
-      log.info(`watching extensions: ${this.#config.watcher.exts.join(",")}`);
+      logger.info(
+        `watching extensions: ${this.#config.watcher.exts.join(",")}`,
+      );
     }
-    log.info("restarting due to changes...");
+    logger.info("restarting due to changes...");
 
     this.killAll();
 
     await this.start();
   }
 
-  private start() {
-    const command = this.#denon.runner.build(this.#script);
-    const process = command.exe();
-    log.debug(`S: starting process with pid ${process.pid}`);
-    this.#processes[process.pid] = (process);
-    this.monitor(process);
-    return command;
+  private async start(): Promise<ScriptOptions> {
+    const commands = this.#denon.runner.build(this.#script);
+
+    // Sequential execution, one process after another is executed,
+    // *sequentially*, the last process is named `main` and is the
+    // one that will actually be demonized.
+    for (let i = 0; i < commands.length; i++) {
+      const plog = log.prefix(`#${i}`);
+      const command = commands[i];
+      let process = command.exe();
+      plog.debug(`starting process with pid ${process.pid}`);
+
+      if (i === commands.length - 1) {
+        plog.warning(`starting main \`${command.cmd.join(" ")}\``);
+        this.#processes[process.pid] = process;
+        this.monitor(process, command.options);
+        return command.options;
+      } else {
```

Comment:
No need for this else block

Reply:
Or rather just remove the `else` keyword and brackets so we save those precious bytes or something lol